### PR TITLE
Fix warning / crash

### LINKED ISSUE #64: Crash when dropping files
With hrev55445, 64bit.

The report:
{{{
Debug information for team /Source/PecoRename/Main/objects.x86_64-cc8-release/PecoRename (3296):
CPU(s): 8x Intel Core™ i7-4712MQ
Memory: 15.92 GiB total, 1.38 GiB used
Haiku revision: hrev55445 Sep 23 2021 07:04:51 (x86_64)

Active Threads:
	thread 3304: w>Select files for renaming 
	thread 3306: w>PecoRename 
	thread 3308: team 3296 debug task 
	thread 3296: PecoRename (main)
		state: Exception (Segment violation)

		Frame		IP			Function Name
		-----------------------------------------------
		0x7fa3739e93c0	0x62091fd113	ConsistencyCheck::CheckForDuplicates() + 0x73 
			Disassembly:
				ConsistencyCheck::CheckForDuplicates():
				0x00000062091fd0a0:               55  push %rbp
				0x00000062091fd0a1:   488b35481f2200  mov 0x221f48(%rip), %rsi
				0x00000062091fd0a8:           4889e5  mov %rsp, %rbp
				0x00000062091fd0ab:             4156  push %r14
				0x00000062091fd0ad:             4155  push %r13
				0x00000062091fd0af:         4c8d6f08  lea 0x8(%rdi), %r13
				0x00000062091fd0b3:             4154  push %r12
				0x00000062091fd0b5:           4c89ef  mov %r13, %rdi
				0x00000062091fd0b8:               53  push %rbx
				0x00000062091fd0b9:             31db  xor %ebx, %ebx
				0x00000062091fd0bb:       e890d3ffff  call 0x62091fa450
				0x00000062091fd0c0:           4c89ef  mov %r13, %rdi
				0x00000062091fd0c3:       e858e7ffff  call 0x62091fb820
				0x00000062091fd0c8:           83f801  cmp $0x1, %eax
				0x00000062091fd0cb:     0f8ed0000000  jle 0x62091fd1a1
				0x00000062091fd0d1:   0f1f8000000000  nop (%rax)
				0x00000062091fd0d8:           4c89ef  mov %r13, %rdi
				0x00000062091fd0db:       e840e7ffff  call 0x62091fb820
				0x00000062091fd0e0:             89de  mov %ebx, %esi
				0x00000062091fd0e2:           4c89ef  mov %r13, %rdi
				0x00000062091fd0e5:           83c301  add $0x1, %ebx
				0x00000062091fd0e8:       e803e7ffff  call 0x62091fb7f0
				0x00000062091fd0ed:             89de  mov %ebx, %esi
				0x00000062091fd0ef:           4c89ef  mov %r13, %rdi
				0x00000062091fd0f2:           4989c4  mov %rax, %r12
				0x00000062091fd0f5:       e8f6e6ffff  call 0x62091fb7f0
				0x00000062091fd0fa:           4989c6  mov %rax, %r14
				0x00000062091fd0fd:         418b0424  mov (%r12), %eax
				0x00000062091fd101:           83f801  cmp $0x1, %eax
				0x00000062091fd104:             747a  jz 0x91fd180
				0x00000062091fd106:             85c0  test %eax, %eax
				0x00000062091fd108:             7546  jnz 0x91fd150
				0x00000062091fd10a:       498b442410  mov 0x10(%r12), %rax
				0x00000062091fd10f:         488d7860  lea 0x60(%rax), %rdi
				0x00000062091fd113:           418b06  mov (%r14), %eax <--

			Frame memory:
				[0x7fa3739e9390]  ...s............   f0 93 9e 73 a3 7f 00 00 c0 8a 00 18 00 00 00 00
				[0x7fa3739e93a0]  ...s....h...C...   c0 93 9e 73 a3 7f 00 00 68 06 bf 19 43 02 00 00
				[0x7fa3739e93b0]  @..s....U...b...   40 94 9e 73 a3 7f 00 00 55 e9 1f 09 62 00 00 00
		0x7fa3739e9450	0x62091fe950	MakeList() + 0xb0 
		0x7fa3739e9630	0x6209200cff	PecoApp::RefsReceived(BMessage*) + 0x4bf 
		0x7fa3739e9840	0x6209201541	PecoApp::MessageReceived(BMessage*) + 0x71 
		0x7fa3739e9a90	0x243197fd0ee	BApplication::DispatchMessage(BMessage*, BHandler*) + 0x1ae 
		0x7fa3739e9ae0	0x243198044f6	BLooper::task_looper() + 0x266 
		0x7fa3739e9b00	0x243197fa2dc	BApplication::Run() + 0x1c 
		0x7fa3739e9d20	0x62091fcd60	main + 0x20 
		0x7fa3739e9d50	0x62091fceb9	_start + 0x39 
		0x7fa3739e9d80	0x142f0a783e3	runtime_loader + 0x103 
		00000000	0x7ffffba54260	commpage_thread_exit + 0 

		Registers:
			  rip:	0x00000062091fd113
			  rsp:	0x00007fa3739e9390
			  rbp:	0x00007fa3739e93b0
			  rax:	0x000000e27bd7ee00
			  rbx:	0x000000000000000a
			  rcx:	0x000000000000006e
			  rdx:	0x0000000000000000
			  rsi:	0x000000000000000a
			  rdi:	0x000000e27bd7ef58
			   r8:	0x0000000000000055
			   r9:	0x000000e27bd5c840
			  r10:	0x000000e27bd7cd00
			  r11:	0x0000000000000246
			  r12:	0x000000e27bd66480
			  r13:	0x00007fa3739e93f8
			  r14:	0x0000000000000000
			  r15:	0x0000000000000005
			   cs:	0x002b
			   ds:	0x0000
			   es:	0x0000
			   fs:	0x0000
			   gs:	0x0000
			   ss:	0x0023
			  st0:	0
			  st1:	0
			  st2:	0
			  st3:	0
			  st4:	0
			  st5:	0
			  st6:	0
			  st7:	0
			  mm0:	{0, 0, 0, 0}
			  mm1:	{0, 0, 0, 0}
			  mm2:	{0, 0, 0, 0}
			  mm3:	{0, 0, 0, 0}
			  mm4:	{0, 0, 0, 0}
			  mm5:	{0, 0, 0, 0}
			  mm6:	{0, 0, 0, 0}
			  mm7:	{0, 0, 0, 0}
			 ymm0:	{0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm1:	{0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0xcccc, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm2:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm3:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm4:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm5:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm6:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm7:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm8:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			 ymm9:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			ymm10:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			ymm11:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			ymm12:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			ymm13:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			ymm14:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
			ymm15:	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}

Loaded Images:
	ID		Text Base	Text End	Data Base	Data End	Type	Name
	--------------------------------------------------------------------------------
	23554	0x1595bc2000	0x1595d8b000	0x1595f8b000	0x1595fa1000	lib    	/boot/system/lib/libicuuc.so.66.1
	23544	0x409f6dc000	0x409f7eb000	0x409f9ea000	0x409fa3c000	lib    	/boot/system/lib/libroot_debug.so
	23557	0x53bf920000	0x53bf935000	0x53bfb34000	0x53bfb36000	lib    	/boot/system/lib/libtranslation.so
	23555	0x59c2be8000	0x59c2c04000	0x59c2e03000	0x59c2e04000	lib    	/boot/system/lib/libz.so.1.2.11
	23553	0x5ed9ca9000	0x5ed9cd9000	0x5ed9ed8000	0x5ed9f0c000	lib    	/boot/system/lib/libicutu.so.66.1
	23548	0x5ee0a55000	0x5ee0bb7000	0x5ee0db6000	0x5ee0dca000	lib    	/boot/system/lib/libstdc++.so.6.0.25
	23543	0x62091e2000	0x620921c000	0x620941c000	0x6209421000	app    	/Source/PecoRename/Main/objects.x86_64-cc8-release/PecoRename
	23556	0xb71934d000	0xb719419000	0xb719618000	0xb719619000	lib    	/boot/system/lib/libzstd.so.1.4.5
	23551	0xbd7a423000	0xbd7a70b000	0xbd7a90a000	0xbd7a922000	lib    	/boot/system/lib/libicui18n.so.66.1
	23550	0xe27d013000	0xe27d014000	0xe27d213000	0xe27d214000	lib    	/boot/system/lib/libicudata.so.66.1
	23549	0x12e27440000	0x12e27455000	0x12e27655000	0x12e27656000	lib    	/boot/system/lib/libgcc_s.so.1
	23560	0x134321c9000	0x134321da000	0x134323da000	0x134323db000	lib    	/boot/system/lib/libroot-addon-icu.so
	23546	0x13464f93000	0x1346508d000	0x1346528d000	0x13465290000	lib    	/boot/system/lib/libtextencoding.so
	23542	0x142f0a66000	0x142f0a85000	0x142f0c84000	0x142f0c86000	lib    	/boot/system/runtime_loader
	23552	0x1c75b645000	0x1c75b651000	0x1c75b850000	0x1c75b852000	lib    	/boot/system/lib/libicuio.so.66.1
	23547	0x1ddd92f2000	0x1ddd94a8000	0x1ddd96a8000	0x1ddd96c2000	lib    	/boot/system/lib/libtracker.so
	23559	0x2225e3b8000	0x2225e3bf000	0x2225e5bf000	0x2225e5c0000	add-on 	/boot/system/add-ons/locale/catalogs/plaintext
	23545	0x243196ad000	0x243199c4000	0x24319bc3000	0x24319bf1000	lib    	/boot/system/lib/libbe.so
	23541	0x7ffffba54000	0x7ffffba5c000	0x00000000	0x00000000	system 	commpage

Areas:
	ID		Base		End			Size (KiB)	Protection	Locking			Name
	--------------------------------------------------------------------------------
	78075	0x18000000	0x18100000	      1024	rw-c       	none          	heap
	78086	0x60000000	0x60001000	         4	rw-cS      	none          	server_memory
	78088	0x68000000	0x68036000	       216	rw-cS      	none          	server_memory
	78061	0x1595bc2000	0x1595d8b000	      1828	r-x        	none          	libicuuc.so.66.1_seg0ro
	78062	0x1595f8b000	0x1595fa0000	        84	rw-        	none          	libicuuc.so.66.1_seg1rw
	78063	0x1595fa0000	0x1595fa1000	         4	rw-        	none          	libicuuc.so.66.1_seg2rw
	78036	0x409f6dc000	0x409f7eb000	      1084	r-x        	none          	libroot_debug.so_seg0ro
	78037	0x409f9ea000	0x409f9f8000	        56	rw-        	none          	libroot_debug.so_seg1rw
	78038	0x409f9f8000	0x409fa3c000	       272	rw-        	none          	libroot_debug.so_seg2rw
	78068	0x53bf920000	0x53bf935000	        84	r-x        	none          	libtranslation.so_seg0ro
	78069	0x53bfb34000	0x53bfb36000	         8	rw-        	none          	libtranslation.so_seg1rw
	78064	0x59c2be8000	0x59c2c04000	       112	r-x        	none          	libz.so.1.2.11_seg0ro
	78065	0x59c2e03000	0x59c2e04000	         4	rw-        	none          	libz.so.1.2.11_seg1rw
	78058	0x5ed9ca9000	0x5ed9cd9000	       192	r-x        	none          	libicutu.so.66.1_seg0ro
	78059	0x5ed9ed8000	0x5ed9edb000	        12	rw-        	none          	libicutu.so.66.1_seg1rw
	78060	0x5ed9edb000	0x5ed9f0c000	       196	rw-        	none          	libicutu.so.66.1_seg2rw
	78046	0x5ee0a55000	0x5ee0bb7000	      1416	r-x        	none          	libstdc++.so.6.0.25_seg0ro
	78047	0x5ee0db6000	0x5ee0dc2000	        48	rw-        	none          	libstdc++.so.6.0.25_seg1rw
	78048	0x5ee0dc2000	0x5ee0dca000	        32	rw-        	none          	libstdc++.so.6.0.25_seg2rw
	78033	0x62091e2000	0x620921c000	       232	r-x        	none          	PecoRename_seg0ro
	78034	0x620941c000	0x6209420000	        16	rw-        	none          	PecoRename_seg1rw
	78035	0x6209420000	0x6209421000	         4	rw-        	none          	PecoRename_seg2rw
	78031	0x655d5cc000	0x655d5dc000	        64	rw-        	none          	rld heap
	78066	0xb71934d000	0xb719419000	       816	r-x        	none          	libzstd.so.1.4.5_seg0ro
	78067	0xb719618000	0xb719619000	         4	rw-        	none          	libzstd.so.1.4.5_seg1rw
	78053	0xbd7a423000	0xbd7a70b000	      2976	r-x        	none          	libicui18n.so.66.1_seg0ro
	78054	0xbd7a90a000	0xbd7a921000	        92	rw-        	none          	libicui18n.so.66.1_seg1rw
	78055	0xbd7a921000	0xbd7a922000	         4	rw-        	none          	libicui18n.so.66.1_seg2rw
	78032	0xcb97f26000	0xcb97f27000	         4	rw-cS      	none          	_rld_debug_
	78070	0xd9d0f4c000	0xd9d0f54000	        32	rw-        	none          	rld heap
	78071	0xe02036b000	0xe02037b000	        64	rw-        	none          	rld heap
	78072	0xe1739db000	0xe1739eb000	        64	rw-        	none          	rld heap
	78073	0xe2062ca000	0xe2062e2000	        96	rw-        	none          	rld heap
	78074	0xe216222000	0xe21623a000	        96	rw-        	none          	rld heap
	78076	0xe2591ef000	0xe2593ef000	      2048	rw-c       	none          	additional heap area
	78077	0xe26d8a4000	0xe26f361000	     27380	r--S       	none          	libicuuc.so.66 mmap area
	78091	0xe278148000	0xe278348000	      2048	rw-c       	none          	additional heap area
	78102	0xe27bd29000	0xe27bf29000	      2048	rw-c       	none          	additional heap area
	78051	0xe27d013000	0xe27d014000	         4	r-x        	none          	libicudata.so.66.1_seg0ro
	78052	0xe27d213000	0xe27d214000	         4	rw-        	none          	libicudata.so.66.1_seg1rw
	78049	0x12e27440000	0x12e27455000	        84	r-x        	none          	libgcc_s.so.1_seg0ro
	78050	0x12e27655000	0x12e27656000	         4	rw-        	none          	libgcc_s.so.1_seg1rw
	78109	0x134321c9000	0x134321da000	        68	r-x        	none          	libroot-addon-icu.so_seg0ro
	78110	0x134323da000	0x134323db000	         4	rw-        	none          	libroot-addon-icu.so_seg1rw
	78041	0x13464f93000	0x1346508d000	      1000	r-x        	none          	libtextencoding.so_seg0ro
	78042	0x1346528d000	0x13465290000	        12	rw-        	none          	libtextencoding.so_seg1rw
	78029	0x142f0a66000	0x142f0a85000	       124	r-x        	none          	runtime_loader_seg0ro
	78030	0x142f0c84000	0x142f0c86000	         8	rw-        	none          	runtime_loader_seg1rw
	78056	0x1c75b645000	0x1c75b651000	        48	r-x        	none          	libicuio.so.66.1_seg0ro
	78057	0x1c75b850000	0x1c75b852000	         8	rw-        	none          	libicuio.so.66.1_seg1rw
	78043	0x1ddd92f2000	0x1ddd94a8000	      1752	r-x        	none          	libtracker.so_seg0ro
	78044	0x1ddd96a8000	0x1ddd96c1000	       100	rw-        	none          	libtracker.so_seg1rw
	78045	0x1ddd96c1000	0x1ddd96c2000	         4	rw-        	none          	libtracker.so_seg2rw
	78080	0x2225e3b8000	0x2225e3bf000	        28	r-x        	none          	plaintext_seg0ro
	78081	0x2225e5bf000	0x2225e5c0000	         4	rw-        	none          	plaintext_seg1rw
	78039	0x243196ad000	0x243199c4000	      3164	r-x        	none          	libbe.so_seg0ro
	78040	0x24319bc3000	0x24319bf1000	       184	rw-        	none          	libbe.so_seg1rw
	78026	0x7f373f4d1000	0x7f373f4d5000	        16	rw-k       	full          	user area
	78027	0x7fa3729e6000	0x7fa3739ec000	     16408	rw-s       	none          	PecoRename_3296_stack
	78106	0x7fdfe9f64000	0x7fdfe9fa9000	       276	rw-s       	none          	w>PecoRename_3306_stack
	78101	0x7ff5d3e84000	0x7ff5d3ec9000	       276	rw-s       	none          	w>TrackerWindow_3304_stack
	78028	0x7ffffba54000	0x7ffffba5c000	        32	r-xSk      	full          	commpage

Protection Flags: r - read, w - write, x - execute, s - stack, o - overcommit, c - cloneable, S - shared, k - kernel

Semaphores:
	ID		Count	Last Holder	Name
	------------------------------------------------------------
	701216	    0	          0	Catalog
	701217	    0	       -990	clipboard
	701218	    0	          0	width buffer
	703969	    0	          0	some BLocker
	704539	    0	       3304	AppServerLink_sLock
	704540	    0	      -1379	token space
	704541	    0	       3306	BLooperList lock
	704843	    0	          0	some BBlockCache lock
	704844	    0	          0	some BLocker
	704845	    0	          0	some BLocker
	704846	    0	          0	Catalog
	704847	    0	          0	LocaleRosterData
	704848	    0	          0	some BLocker
	704849	    0	          0	Catalog
	704850	    0	          0	some BLocker
	704851	    0	          0	some BLocker
	704852	    0	          0	undo
	704853	    0	          0	Light Lock
	704854	    0	          0	Light Lock
	704855	    0	          0	PeriodicUpdatePoses
	704856	    0	          0	BMessageQueue Lock
	704857	    0	          0	AppLooperPort
	704872	    0	          0	screen list
	704880	    0	          0	BMessageQueue Lock
	704881	   -1	       3296	PecoRename
	704887	    0	          0	BMessageQueue Lock
	704888	    0	          0	offscreen bitmap
	704895	    0	          0	BMessageQueue Lock
	704896	    0	          0	offscreen bitmap
	704906	    0	          0	BMessageQueue Lock
	704907	    0	       3304	TrackerWindow
	704913	    0	          0	Tracker node icon cache
	704914	    0	          0	Tracker shared icon cache
	704927	    0	          0	some BLocker
	704929	    0	          0	virtual directory manager
	704930	    0	          0	Catalog
	704993	    0	          0	tool tip manager
}}}

### PR DESCRIPTION
**Pull request by humdingerb**
Do return something when returning from non-void functions.
Fixes #64

Since I'm just a (now rusty) dabbler: Is this even the right fix for #64. Also, is it OK to return NULL in NameToCheck().Name()?

### CODE REVIEW

File: Main/ConsistencyCheck.cpp [hunk 1]
```diff
@@ -88,6 +89,8 @@ ConsistencyCheck::CheckForDuplicates()
 		if (name1->Name().Compare(name2->Name()) == 0) {
 			name1->SetDuplicate();
 			name2->SetDuplicate();
+			duplicateFound = true;
 		}
 	}
+	return duplicateFound;
```
Comment (scottmc) [inline Main/ConsistencyCheck.cpp:95]: I think you can do this without creating duplicateFound.  Just return true on line 92, or return false on 95.

**Comment by scottmc** [comment]
Check it using multiple items to make sure it acts on all of them as expected.  I haven't studied the code, i assumed it was being called from a loop, if so that it would likely work.  It checks to make sure you aren't trying to rename it to the same name as an already existing file right?  You should check that as well.  Make sure it errors out correctly.

I'm unsure about the second part of the PR, whether the NULL return is correct in this case.

**Comment by scottmc** [comment]
> Still not sure about the NULL return either...

the function is supposed to return a BString.  The switch has three cases covered.  Are there any other possible cases that should be covered?  If not and you are just trying to stop a warning...  you might just add a default case at the end and have it return something.  Sometimes compilers freak out because there are uncovered cases, having a default covers that.  Maybe NULL is ok, or maybe an empty string, or "NOT_A_FILE" or something else?

File: Main/ConsistencyCheck.h [hunk 2]
```diff
@@ -31,6 +31,7 @@ class NameToCheck {
 					case item_new_name:	return fFileListItem->fNewName;
 					case item_disk_name: return fName;
 				}
+				return NULL;
```
Comment (korli) [inline Main/ConsistencyCheck.h:34]: you can replace "return NULL;" with "abort();" 

it should fix the warning because abort() never returns

**Comment by humdingerb** [comment]
That should fix the crashing.

**Pull request merged**